### PR TITLE
RATEST-150S:New E2E Test For FindPatient Workflow

### PR DESCRIPTION
**Pull request by sherrif10**
https://issues.openmrs.org/browse/RATEST-221

### CODE REVIEW

**Comment by sherrif10** [comment]
Hello @kdaud  Waiting on your reviews

**Comment by sherrif10** [comment]
Waiting on your reviews @kdaud  to finalise this workflow

File: src/test/java/org/openmrs/contrib/qaframework/RunTest.java
Comment (kdaud) [inline src/test/java/org/openmrs/contrib/qaframework/RunTest.java:48]: What do you think of `SELENIUM_FIND_PATTIENT`?

File: src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java
Comment (kdaud) [inline src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java:123]: Is this change necessary?

Comment (sherrif10) [reply src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java:123]: This is breaking my test workflow ,so I decided to fix it in here

File: src/test/java/org/openmrs/contrib/qaframework/RunTest.java
Comment (sherrif10) [reply src/test/java/org/openmrs/contrib/qaframework/RunTest.java]: Resolved thanks

File: src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java
Comment (kdaud) [reply src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java:123]: > This is breaking my test workflow ,so I decided to fix it in here

@sherrif10 did you get a chance to have a look  at  [RATEST-232](https://issues.openmrs.org/browse/RATEST-232)?

Comment (sherrif10) [reply src/test/java/org/openmrs/contrib/qaframework/automation/VitalsAndTriagingSteps.java:123]: > did you get a chance to have a look at [RATEST-232](https://issues.openmrs.org/browse/RATEST-232)?

Sure , but still I could resolve it from here since it was a simple fix that make the test to pass

**Comment by sherrif10** [comment]
cc @kdaud  more reviews or approving

**Comment by kdaud** [comment]
Why not renaming your `.yml `file to `refapp-2x-find-patient.yml` as per the naming of the existing workflow [files](https://github.com/openmrs/openmrs-contrib-qaframework/tree/master/.github/workflows)?

**Comment by kdaud** [comment]
Worth updating your branch with master

**Comment by sherrif10** [comment]
This pull request has no merge conflicts

**Comment by sherrif10** [comment]
Waiting for your final reviews, i hope this is ready to get merged

File: README.md
Comment (kdaud) [inline README.md:30]: As the .yml file is renamed the stattus buldge markdown changes. I guess to be now

` [RefApp 2.x Find Patient](https://github.com/openmrs/openmrs-contrib-qaframework/actions/workflows/refapp-2.x-find-Patient.yml)`

Comment (sherrif10) [reply README.md:30]: Is there a  difference than the above change

**Comment by kdaud** [comment]
Thanks @sherrif10 for the work!

As you catch my last review [above](https://github.com/openmrs/openmrs-contrib-qaframework/pull/127#discussion_r710086447), I have two things to shade to you

1. The way you have renamed the .yml file shouts loud than the existing file name.
2. Would it be a good idea you create a ticket and work around renaming those .yml files for 2.x and 3.x refapp workflows to look like the one in this PR?

File: README.md
Comment (kdaud) [reply README.md:30]: > @kdaud i didnt get your point above , Is there a difference than the above chang


As the .yml was renamed from refapp-2x-findPatient.yml --> refapp-2x-find-patient.yml, the status badge mark down changed as well from `[! [RefApp 2.x Find-Patient].....](),` to `[! [RefApp 2.x Find Patient].....](),`

Comment (sherrif10) [reply README.md]: Nice catch Resolved thanks

**Comment by sherrif10** [comment]
> 2\. Would it be a good idea you create a ticket and work around renaming those .yml files for 2.x and 3.x refapp workflows to look like the one in this PR?  eg  `refapp-2x-clinical-visit.yml` to `refapp-2.x-clinical-visit.yml`, re`fapp-3x-user-setting.yml` to `refapp-3.x-user-setting.yml`

Will fix this in this very pull request

**Comment by kdaud** [comment]
> Sure we can handle that in the separate pull request

@sherrif10 I agree! Would be great if you work it out!

**Comment by sherrif10** [comment]
> Would be great if you work it out!

Sure no problem , Let me quickly handle that.

**Comment by sherrif10** [comment]
Any more reviews or approving @kdaud

**Review by kdaud — APPROVED** [review]
> Any more reviews

Apart from approval !

Its high time this commit enters into master, unless @dkayiwa has one or two reviews to take, But I guess he will trigger the merge button for the magic to happen.

File: src/test/java/org/openmrs/contrib/qaframework/automation/FindPatientSteps.java
Comment (dkayiwa) [inline src/test/java/org/openmrs/contrib/qaframework/automation/FindPatientSteps.java:49]: `enterPatientIdentifier`

Comment (sherrif10) [reply src/test/java/org/openmrs/contrib/qaframework/automation/FindPatientSteps.java]: Resolved Thanks @dkayiwa

**Comment by kdaud** [comment]
@sherrif10 worth looking into the status badge for the test to have it visible on the dashboard. Look through in the ReadMe.md file to find out.

**Comment by sherrif10** [comment]
looking into it thnks

**Comment by kdaud** [comment]
> looking into it thnks

How about making a grand PR that captures as well the idea [here](https://github.com/openmrs/openmrs-contrib-qaframework/pull/127#issuecomment-920878050) !

**Comment by sherrif10** [comment]
Lets make it so simple as  refapp-2x- not refapp2.x , since all workflows are written as refapp-2x

**Comment by kdaud** [comment]
> Lets make it so simple as refapp-2x- not refapp2.x , since all workflows are written as refapp-2x

I though it would look like refapp-2.x- not refapp-2x- . What do you think?

**Comment by sherrif10** [comment]
workl
All workflows are written as refapp-2x not refapp-2.x ,if that is so Lets have a different ticket to change all the workflows to refapp-2.x.

**Comment by kdaud** [comment]
> All workflows are written as refapp-2x not refapp-2.x

@sherrif10 think fully maintaining the original naming is sweet!

**Pull request merged**
